### DISCUSSION TITLE
`estimateGas` always throws `RpcRequestError` when `chainId = 324` (zksync era)

### DISCUSSION THREAD

**Discussion by chloe-tan** [wevm/wagmi#4016] [Q&A]
Running a simple `estimateGas` function always fails with an `RpcRequestError` when chainId = **324** (zksync era). 
- When chainId is anything else (e.g. 1, 137, 8453, ...) everything works fine.
- RPCs i've tried with: https://mainnet.era.zksync.io, https://zksync.drpc.org

Wondering if anyone is encountering the same issue and what a workaround might be. Thank you!

```ts

import { useConfig } from 'wagmi'
import { estimateGas } from 'wagmi/core'

// In some react component:

const wagmiConfig = useConfig()

const gasEstimateResult = await estimateGas(wagmiConfig, {
   to: SOME_ADDRESS,
   chainId: SOME_CHAIN_ID,
   value: SOME_VALUE,
})
```

Error:

<img width="469" alt="image" src="https://github.com/wevm/wagmi/assets/95644202/dbd02a81-389d-4902-9c82-e94042c4a783">

**Comment by flymauntains** [comment]
I also same error.
Do y have any lead about error

**Reply by chloe-tan** [reply]
Unfortunately no :/ I've also noticed gas estimation simulation for zksync on wallet browser extensions like rainbow and metamask are also failing. Seems like the fundamental eth_estimateGas primitive on zksync rpcs/the chain itself is not working well. My intuition tells me setting a higher gas limit might help, but i've not verified that yet.

**Reply by flymauntains** [reply]
Sorry. what is  wagmi version in your project?

**Reply by chloe-tan** [reply]
^2.9.2

**Reply by flymauntains** [reply]
When build rainbowkit opensource, need rainbow_provider_api_key.
Where get it?
